pkg/plugins/autodiscovery/cargo/dependencyManifest.go

```diff
@@ -40,17 +40,83 @@ sources:
       Key: '{{ .ExistingSourceKey }}'
 conditions:
   {{ .ConditionID }}:
-    name: 'Ensure Cargo chart named "{{ .DependencyName }}" is specified'
+    name: 'Test if version of "{{ .DependencyName }}" {{"{{"}} source "{{ .ExistingSourceID }}" {{"}}"}} differs from {{"{{"}} source "{{ .SourceID }}" {{"}}"}}'
+    kind: 'shell'
+    sourceid: '{{ .SourceID }}'
+{{- if .ScmID }}
+    scmid: '{{ .ScmID }}'
+{{- end }}
+    spec:
+      command: 'test {{"{{"}} source "{{ .ExistingSourceID }}" {{"}}"}} != '
+targets:
+  {{ .TargetID }}:
+    name: '{{ .TargetName }}'
+    kind: 'shell'
+{{- if .ScmID }}
+    scmid: '{{ .ScmID }}'
+{{- end }}
+    spec:
+      command: |
+        ARGS=""
+        if [ "$DRY_RUN" = "true" ]; then
+          ARGS="--dry-run"
+        fi
+        cargo upgrade $ARGS --manifest-path {{ .File }} --package {{ .DependencyName }}@{{"{{"}} source "{{ .SourceID }}" {{"}}"}}
+        cargo update $ARGS --manifest-path {{ .File }} {{ .DependencyName }}@{{"{{"}} source "{{ .SourceID }}" {{"}}"}} --precise {{"{{"}} source "{{ .SourceID }}" {{"}}"}}
+      changedif:
+        kind: file/checksum
+        spec:
+          files:
+            - "{{ .File }}"
+            - "Cargo.lock"
```

Comment:
Is the `Cargo.lock` file always at the same location? Shouldn't it be next to the Cargo.toml file?

Reply:
I've done a better rewrite where I handle this correctly